### PR TITLE
`data.azurerm_api_management` - Support  property `tenant_access`

### LINKED ISSUE #19406: Data Source: azurerm_api_management missing tenant_access information
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2

### AzureRM Provider Version

3.32

### Affected Resource(s)/Data Source(s)

azurerm_api_management

### Terraform Configuration Files

```hcl
data "azurerm_api_management" "apim" {
  name                = "swo-ip-apim-${var.environment}"
  resource_group_name = var.resource_group_name
}

resource "azurerm_key_vault_secret" "apim_management_key" {
  name         = "APIM-ManagementKey"
  value        = data.azurerm_api_management.apim.tenant_access[0].primary_key
  key_vault_id = var.keyvault_ressource_id
}
```

### Debug Output/Panic Output

```shell
n/a
```

### Expected Behaviour

Data source `azurerm_api_management` should export a `tenant_access` information block.

### Actual Behaviour

When accessing the data source `azurerm_api_management` the following error occurs: 

This object has no argument, nested block, or exported attribute named "tenant_access".

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

The API Management Rest API has an valid endpoint to get those information:

https://learn.microsoft.com/en-us/rest/api/apimanagement/current-ga/tenant-access/get?tabs=HTTP

### PR DESCRIPTION
**Pull request by sinbai**
Fix #19406.

Test result:
PASS: TestAccDataSourceApiManagement_tenantAccess (4848.68s)

### CODE REVIEW

File: internal/services/apimanagement/api_management_data_source.go [hunk 1]
```diff
@@ -263,6 +291,16 @@ func dataSourceApiManagementRead(d *pluginsdk.ResourceData, meta interface{}) er
...
 
+	if resp.Sku.Name != apimanagement.SkuTypeConsumption {
+		tenantAccessInformationContract, err := tenantAccessClient.ListSecrets(ctx, id.ResourceGroup, id.ServiceName, "access")
+		if err != nil {
+			return fmt.Errorf("retrieving tenant access properties for %s: %+v", *id, err)
+		}
+		if err := d.Set("tenant_access", flattenApiManagementTenantAccessSettings(tenantAccessInformationContract)); err != nil {
+			return fmt.Errorf("setting `tenant_access`: %+v", err)
+		}
+	}
```
Comment (tombuildsstuff) [inline internal/services/apimanagement/api_management_data_source.go:302]: we need to ensure the `tenant_access` block is always set into the state:

```suggestion
	tenantAccess := make([]interface{}, 0)
	if resp.Sku.Name != apimanagement.SkuTypeConsumption {
		tenantAccessInformationContract, err := tenantAccessClient.ListSecrets(ctx, id.ResourceGroup, id.ServiceName, "access")
		if err != nil {
			return fmt.Errorf("retrieving tenant access properties for %s: %+v", *id, err)
		}
		tenantAccess = flattenApiManagementTenantAccessSettings(tenantAccessInformationContract)
	}
	if err := d.Set("tenant_access", tenantAccess); err != nil {
		return fmt.Errorf("setting `tenant_access`: %+v", err)
	}
```

File: internal/services/apimanagement/api_management_data_source_test.go [hunk 2]
```diff
@@ -28,6 +28,18 @@ func TestAccDataSourceApiManagement_basic(t *testing.T) {
...
 }
 
+func TestAccDataSourceApiManagement_tenantAccess(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_api_management", "test")
+	r := ApiManagementDataSource{}
+
+	data.DataSourceTest(t, []acceptance.TestStep{
+		{
+			Config: r.tenantAccess(data),
+			Check:  acceptance.ComposeTestCheckFunc(),
```
Comment (tombuildsstuff) [inline internal/services/apimanagement/api_management_data_source_test.go:38]: this isn't checking anything, let's check that we have some values here:

```suggestion
			Check:  acceptance.ComposeTestCheckFunc(
				check.That(data.ResourceName).Key("tenant_access.0.enabled").Exists(),
			),
```

we should also update the `basic` test above to ensure there's no items if this isn't consumption:

> 			check.That(data.ResourceName).Key("tenant_access.#").HasValue("0"),

File: website/docs/d/api_management.html.markdown [hunk 3]
```diff
@@ -177,6 +179,18 @@ A `sku` block exports the following:
 
 ---
 
+A `tenant_access` block supports the following:
```
Comment (tombuildsstuff) [inline website/docs/d/api_management.html.markdown:182]: this is read-only:

```suggestion
A `tenant_access` block exports the following:
```

File: website/docs/d/api_management.html.markdown [hunk 4]
```diff
@@ -177,6 +179,18 @@ A `sku` block exports the following:
...
+
+* `enabled` - Should the access to the management API be enabled?
```
Comment (tombuildsstuff) [inline website/docs/d/api_management.html.markdown:184]: this is read-only:

```suggestion
* `enabled` - Is access to the Management API enabled (presumably "for this Tenant")?
```

File: website/docs/d/api_management.html.markdown [hunk 5]
```diff
@@ -177,6 +179,18 @@ A `sku` block exports the following:
...
+
+* `tenant_id` - The identifier for the tenant access information contract.
```
Comment (tombuildsstuff) [inline website/docs/d/api_management.html.markdown:186]: presumably:

```suggestion
* `tenant_id` - The ID of the Tenant which has access to this API Management instance.
```

?

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hi @sinbai 

Thanks for this PR - I've taken a look through and left a couple of comments inline, if we can fix those up then we should be able to take another look here.

Thanks!

File: internal/services/apimanagement/api_management_data_source.go (hunk 1 above)
Comment (sinbai) [reply internal/services/apimanagement/api_management_data_source.go]: Thanks for you feedback. Fixed.

File: internal/services/apimanagement/api_management_data_source_test.go (hunk 2 above)
Comment (sinbai) [reply internal/services/apimanagement/api_management_data_source_test.go]: Thanks for you feedback. For the basic test, In fact the sku name is not consumption should have `tenant_access`. So I added `check.That(data.ResourceName).Key("tenant_access.#").HasValue("1")` for it. Please let me know if I missed something. Thank you.

File: website/docs/d/api_management.html.markdown (hunk 3 above)
Comment (sinbai) [reply website/docs/d/api_management.html.markdown]: Thanks for you feedback. Fixed.

File: website/docs/d/api_management.html.markdown (hunk 5 above)
Comment (sinbai) [reply website/docs/d/api_management.html.markdown]: Thanks for you feedback. Fixed.

File: website/docs/d/api_management.html.markdown (hunk 4 above)
Comment (sinbai) [reply website/docs/d/api_management.html.markdown]: Thanks for you feedback. Fixed.

**Review by stephybun — APPROVED** [review]
Thanks @sinbai, LGTM 🦚

**Pull request merged**